src/zc/buildout/easy_install.py

```diff
@@ -213,20 +233,34 @@ def __init__(self,
 
         self._index_url = index
         path = (path and path[:] or []) + buildout_and_setuptools_path
-        if dest is not None and dest not in path:
-            path.insert(0, dest)
         self._path = path
         if self._dest is None:
             newest = False
         self._newest = newest
-        self._env = pkg_resources.Environment(path)
         self._index = _get_index(index, links, self._allow_hosts)
         self._requirements_and_constraints = []
         self._check_picked = check_picked
 
         if versions is not None:
             self._versions = normalize_versions(versions)
 
+    @reify
+    def _env(self):
+        return pkg_resources.Environment(self._get_full_path())
+
+    def _get_full_path(self):
+        return ([self._dest] if self._dest is not None else []) + self._path
+
+    def _env_rescan_dest(self):
+        self._env.scan([self._dest])
+
+    def _env_get_best_match(self, requirement, ws):
+        dist = self._env.best_match(requirement, ws)
+        return dist
+
+    def _env_get_dists_for_project(self, project_name):
+        return self._env[project_name]
+
```

Comment:
> I realize that these are hooks. But It annoys me to add more levels of call, mainly because it makes the code harder to read.

It's more than just hooks. By wrapping all access to `._env` I can make sure all it's inputs and outputs are interceptable in subclasses of Installer without creating a subclass of Environment.

For installing wheels, I can get away with intercepting either only the places that modify `._env` or the places that read from it, noot necessarily both. But if I leave some `._env` access unwrapped, there's a risk another direct `._env` access is introduced later that will let non-egg-precedence dists leak in where we're expecting egg-precedence dists.

Anyway, for now, I'll wrap all write accesses to `._env`, which should be enough for `buildout.wheel`.

> Please move the implementations from the humpty-had-a-great-fall branch of buildout.wheel here and inline the small functions that are only called once.

All of them? Including the ones that try to scan for non-egg dists inside `._dest`? That will make `buildout.wheel` tiny. I'll assume silence is consent ;-)